core/services/job/spawner.go

```diff
@@ -168,6 +168,12 @@ func (js *spawner) StartService(ctx context.Context, spec Job) error {
 	// that it was able to start without an error.
 	aj := activeJob{delegate: delegate, spec: spec}
 
+	spec.PipelineSpec.JobName = spec.Name.ValueOrZero()
+	spec.PipelineSpec.JobID = spec.ID
```

Comment:
Well this certainly looks better, are we 100% sure it got loaded from the DB properly?

Reply:
This function might be clearer if `spec` was renamed to `jb`

Reply:
> Well this certainly looks better, are we 100% sure it got loaded from the DB properly?

I am relying on the tests, including manual verification in operator UI. But I can add a temp debug output here verify this concern.